Namespace aliasing causes an exception to be thrown

Hi there,

By default, System.IO namespace is added to every script.

Problem is I have an assembly that defines Fluent.IO.Path that conflicts with System.IO.Path.

So, I'd need to write something like this:

using Path=Fluent.IO.Path;

However, when I try to run my script, I get the following error:
"error CS0246: The type or namespace name 'Path' could not be found (are you missing a using directive or an assembly reference?)"

Any help would be great.

Running scriptcs in debug mode, I can confirm the assembly is being found and loaded.

Thank you,
Brian

https://groups.google.com/forum/m/?fromgroups#!topic/scriptcs/7YiOeeGgaZw

Hi @glennblock , unfortunately, no, the fully qualified reference doesn't work either.

     var inline = PreMailer.Net.PreMailer.MoveCssInline(layout, true);
     File.WriteAllText("Layout.Inline.html", inline.Html);

Produces error:

`error CS0103: The name 'PreMailer' does not exist in the current context`

I'm using latest version on choco v0.12

Brian I submitted a fix several weeks ago to not recursively walk dirs
which was merged into Dev.

On Sat Jan 24 2015 at 10:28:19 PM Brian Chavez <notifications@github.com>
wrote:

> Hi @glennblock <https://github.com/glennblock> , unfortunately, no, the
> fully qualified reference doesn't work either.
>
>  var inline = PreMailer.Net.PreMailer.MoveCssInline(layout, true);
>  File.WriteAllText("Layout.Inline.html", inline.Html);
>
> Produces error:
>
> error CS0103: The name 'PreMailer' does not exist in the current context
>
> I'm using latest version on choco v0.12
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/scriptcs/scriptcs/issues/826#issuecomment-71355806>.
>

--- a/src/ScriptCs.Core/UsingLineProcessor.cs
+++ b/src/ScriptCs.Core/UsingLineProcessor.cs
@@ -30,7 +30,7 @@ namespace ScriptCs
 
         private static bool IsUsingLine(string line)
         {
-            return line.Trim(' ').StartsWith(UsingString) && !line.Contains("{") && line.Contains(";");
+            return line.Trim(' ').StartsWith(UsingString) && !line.Contains("{") && line.Contains(";") && !line.Contains("=");
         }
 
         private static string GetNamespace(string line)

--- a/test/ScriptCs.Core.Tests/UsingLineProcessorTests.cs
+++ b/test/ScriptCs.Core.Tests/UsingLineProcessorTests.cs
@@ -37,6 +37,19 @@ namespace ScriptCs.Tests
             }
 
             [Theory, ScriptCsAutoData]
+            public void ShouldIgnoreAliases(IFileParser parser, UsingLineProcessor processor)
+            {
+                // Arrange
+                const string UsingLine = @"using Path = ""System.IO.Path"";";
+
+                // Act
+                var result = processor.ProcessLine(parser, new FileParserContext(), UsingLine, true);
+
+                // Assert
+                result.ShouldBeFalse();
+            }
+
+            [Theory, ScriptCsAutoData]
             public void ShouldAddNamespaceToContext(IFileParser parser, UsingLineProcessor processor)
             {
                 // Arrange

